.github/workflows/release-tag.yml

```diff
@@ -0,0 +1,28 @@
+name: Tag the new release
+on:  
+  push:
+    tags: ["v*"]
+
+jobs:
+  build:
+    permissions:
+      contents: write
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: |
+          echo "GITHUB_REF is $GITHUB_REF"
+
+          [[ "$GITHUB_REF" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]] || (echo "$GITHUB_REF does not seem like a valid tag. expected format refs/tags/v10.2.3" && exit 1)
```

Comment:
This will error out when we've repushed the major tag (eg `v1`), right?  Instead of erroring out, shall we add a bit of logic to handle this case (no-op, success)?

Or, would we want to update the tag regex above to only trigger if the tag matches eg `v[0-9]+\.[0-9]+\.[0-9]`?

Reply:
that is a valid point. however in my tests it didn't happen. Now I am thinking why :)

<img width="667" alt="Screenshot 2023-03-28 at 10 11 52 PM" src="https://user-images.githubusercontent.com/612092/228309878-02d69892-bec8-41f0-96ef-5eac5dca4297.png">

Reply:
That run looks like it is from the `v2.0.10` tag push, which looks to work as intended.

When `v2` was pushed (at the end of this action), did it trigger another run of this workflow -- and did it succeed?  Or no, this workflow wasn't triggered -- and that's what you're looking into?

Reply:
no it didn't trigger another job, and it kind of surprises me too. but to be on safer side i have updated the action, as you suggested, to trigger when it matches `v*.*.*` pattern.

Reply:
Oh actually, I remember now! If the event source is from a workflow itself, it won't trigger a new one (https://stackoverflow.com/a/69063453).

I do like the update to match the `v*.*.*` pattern, though, in case one of us pushes a major tag 👍